src/proto/server.rs

```diff
@@ -56,6 +62,28 @@ impl ServerState {
         self.server_callbacks
             .insert(callback_type, Box::new(action));
     }
+
+    pub fn create_rpc(
+        &mut self,
+        data: EntryData,
+        callback: impl Fn(Vec<u8>) -> Vec<u8> + Send + RefUnwindSafe + 'static,
+    ) {
+        let id = self.create_entry(data).try_next().unwrap().unwrap();
+        self.rpc_actions.insert(id, Box::new(callback));
+    }
+
+    pub fn call_rpc(&self, id: u16, parameter: Vec<u8>) -> Vec<u8> {
+        let entry = self.entries.get(&id).unwrap();
+        let rpc = self.rpc_actions.get(&id).unwrap();
+
+        match entry.value {
+            EntryValue::RpcDefinition(_) => match panic::catch_unwind(|| rpc(parameter)) {
```

Comment:
Evaluate the RPC body asynchronously, as with the client callback

Reply:
fixed